### PR TITLE
bump pj-on-kind.sh to latest version

### LINKED ISSUE #27676: pj-on-kind.sh installs very old kind version
https://github.com/kubernetes/test-infra/blob/3d45e3f5456af6495dc66c7ec0a0bc53fe8b15b3/prow/pj-on-kind.sh#L82

we should update that

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #27676

### CODE REVIEW

**Comment by BenTheElder** [comment]
cc @chaodaiG @aojea

**Review by chaodaiG — APPROVED** [review]

File: prow/pj-on-kind.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ function ensureInstall() {
   # Install kind and set up cluster if not already done.
   if ! command -v kind >/dev/null 2>&1; then
     echo "Installing kind..."
-    GO111MODULE="on" go get sigs.k8s.io/kind@v0.7.0
+    GO111MODULE="on" go get sigs.k8s.io/kind@v0.16.0
```
Comment (aojea) [inline prow/pj-on-kind.sh:82]: what's this used for?

Comment (aojea) [reply prow/pj-on-kind.sh:82]: it should be `go install` , no?

Comment (BenTheElder) [reply prow/pj-on-kind.sh:82]: possibly, it depends on the go version, for a long time go supported both.
this is used when the user doesn't have kind installed.

the script is used to run prowjobs locally on kind for job author devel

**Pull request merged**